pkg/controllers/machine/disruption/drift.go

```diff
@@ -82,30 +84,42 @@ func (d *Drift) Reconcile(ctx context.Context, provisioner *v1alpha5.Provisioner
 
 // isDrifted will check if a machine is drifted from the fields in the provisioner.Spec and
 // the cloudprovider
-func (d *Drift) isDrifted(ctx context.Context, provisioner *v1alpha5.Provisioner, machine *v1alpha5.Machine) (bool, error) {
-	cloudProviderDrifted, err := d.cloudProvider.IsMachineDrifted(ctx, machine)
+func (d *Drift) isDrifted(ctx context.Context, provisioner *v1alpha5.Provisioner, machine *v1alpha5.Machine) (string, error) {
+	driftedReason, err := d.cloudProvider.IsMachineDrifted(ctx, machine)
 	if err != nil {
-		return false, err
+		return string(NotDrifted), err
 	}
 
-	return cloudProviderDrifted || areStaticFieldsDrifted(provisioner, machine) || areNodeRequirementsDrifted(provisioner, machine), nil
+	reason := lo.FindOrElse([]cloudprovider.DriftReason{driftedReason, areStaticFieldsDrifted(provisioner, machine), areNodeRequirementsDrifted(provisioner, machine)}, "", func(i cloudprovider.DriftReason) bool {
```

Comment:
Shouldn't this just be like this? The enum should be defined in this package. 
```suggestion
	reason := lo.FindOrElse([]DriftReason{driftedReason, areStaticFieldsDrifted(provisioner, machine), areNodeRequirementsDrifted(provisioner, machine)}, "", func(i DriftReason) bool {
```